cli/azd/cmd/infra_create.go

```diff
@@ -74,241 +65,58 @@ func (ica *infraCreateAction) Run(ctx context.Context, cmd *cobra.Command, args
-		// project.
-		selected, err := console.PromptLocation(ctx, "Please select an Azure location to use to store deployment metadata:")
-		if err != nil {
-			return fmt.Errorf("prompting for deployment metadata region: %w", err)
-		}
-
-		location = selected
+		return fmt.Errorf("creating provisioning manager: %w", err)
 	}
 
-	formatter, err := output.GetFormatter(cmd)
+	previewResult, err := infraManager.Preview(ctx)
 	if err != nil {
-		return err
-	}
-	interactive := formatter.Kind() == output.NoneFormat
-
-	// Do the creating. The call to `DeployToSubscription` blocks until the deployment completes,
-	// which can take a bit, so we typically do some progress indication.
-	// For interactive use (default case, using table formatter), we use a spinner.
-	// With JSON formatter we emit progress information, unless --no-progress option was set.
-	deploymentTarget := bicep.NewSubscriptionDeploymentTarget(azCli, location, env.GetSubscriptionId(), env.GetEnvName())
-
-	type deployFuncResult struct {
-		Result azcli.AzCliDeployment
-		Err    error
+		return fmt.Errorf("previewing deployment: %w", err)
 	}
-	var res deployFuncResult
 
-	deployAndReportProgress := func(spinner *spin.Spinner) error {
-		deployResChan := make(chan deployFuncResult)
-		go func() {
-			res, err := bicep.Deploy(ctx, deploymentTarget, bicepPath, azdCtx.BicepParametersFilePath(ica.rootOptions.EnvironmentName, "main"))
-			deployResChan <- deployFuncResult{Result: res, Err: err}
-			close(deployResChan)
-		}()
-
-		progressDisplay := provisioning.NewProvisioningProgressDisplay(infra.NewAzureResourceManager(azCli), env.GetSubscriptionId(), env.GetEnvName())
-
-		for {
-			select {
-			case deployRes := <-deployResChan:
-				res = deployRes
-				return deployRes.Err
-			case <-time.After(10 * time.Second):
-				if ica.noProgress {
-					continue
-				}
-				if interactive {
-					progressDisplay.ReportProgress(ctx, spinner.Title, spinner.Println)
-				} else {
-					reportDeploymentStatusJson(ctx, azCli, env, formatter, cmd)
-				}
-			}
-		}
-	}
-
-	if interactive {
-		deploymentSlug := azure.SubscriptionDeploymentRID(env.GetSubscriptionId(), env.GetEnvName())
-		deploymentURL := withLinkFormat(
-			"https://portal.azure.com/#blade/HubsExtension/DeploymentDetailsBlade/overview/id/%s\n\n",
-			url.PathEscape(deploymentSlug))
-		printWithStyling(
-			"Provisioning Azure resources can take some time.\n\nYou can view detailed progress in the Azure Portal:\n%s",
-			deploymentURL)
-
-		spinner := spin.NewSpinner("Creating Azure resources")
-		spinner.Start()
-		err = deployAndReportProgress(spinner)
-		spinner.Stop()
-
-		if err == nil {
-			fmt.Println("Created Azure resources")
-		}
-	} else {
-		err = deployAndReportProgress(nil)
+	provisioningScope := provisioning.NewSubscriptionScope(ctx, env.GetLocation(), env.GetSubscriptionId(), env.GetEnvName())
+	deployResult, err := infraManager.Deploy(ctx, &previewResult.Deployment, provisioningScope)
```

Comment:
This will likely need to be tweaked as we build out the terraform implementation.  Going to keep as is for now since I've swapped the names multiple times at this point.